index.js

```diff
@@ -21,14 +20,48 @@ class DoxymeMediaAccess {
         audio: this.userMediaStatus.hasMicrophone,
         video: this.userMediaStatus.hasCamera
       }).then(stream => {
-        return waitForDeviceInfo().then(userMediaStatus => {
-          this.userMediaStatus = userMediaStatus;
-
-          const speech = hark(stream, {
-            interval: 500
-          });
-          speech.on('volume_change', volume => {
-            emitter.emit('localVolumeChange', volume);
+        
+        return waitForDeviceInfo().then( () => {
+          navigator.mediaDevices.getUserMedia({ audio: true, video: true })
+          .then(function(stream) {
+            let audioContext, analyser, microphone, javascriptNode;
+            var contextClass = (window.AudioContext || 
+              window.webkitAudioContext || 
+              window.mozAudioContext || 
+              window.oAudioContext || 
+              window.msAudioContext);
+              if (contextClass) {
+                  // Web Audio API is available.
+                  audioContext = new contextClass();
+              } else {
+                  // Web Audio API is not available. Fallback
+              }
+      
+            analyser = audioContext.createAnalyser();
+            microphone = audioContext.createMediaStreamSource(stream);
+            javascriptNode = audioContext.createScriptProcessor(2048, 1, 1);
+          
+            analyser.smoothingTimeConstant = 0.8;
+            analyser.fftSize = 1024;
+          
+            microphone.connect(analyser);
+            analyser.connect(javascriptNode);
+            javascriptNode.connect(audioContext.destination);
+            javascriptNode.onaudioprocess = function() {
+                var array = new Uint8Array(analyser.frequencyBinCount);
+                analyser.getByteFrequencyData(array);
+                var values = 0;
+                var length = array.length;
+                for (var i = 0; i < length; i++) {
+                  values += (array[i]);
+                }
+                var average = values / length;
+                emitter.emit('localVolumeChange', Math.round(average));
+            }
+            })
+            .catch(function(err) {
+              console.error(err);
+              /* handle the error */
```

Comment:
Briefly, what was happening to hark on ios and why did you have to reimplement it from scratch?

Reply:
Hark JS, don't work w/ android, it also has some strange metric system w/ minus values, that can't be properly processed. Overall, it looks like BlackBox with limited customization possibilities. Also, I want to try it to test from the staging, because it's complicated to build and integrate several services locally.